forge-game/src/main/java/forge/game/ability/effects/DestroyAllEffect.java

```diff
@@ -76,7 +79,10 @@ public void resolve(SpellAbility sa) {
         if (sa.hasParam("RememberAllObjects")) {
             card.addRemembered(list);
         }
-
+        
+        if (isOptional && !sa.getActivatingPlayer().getController().confirmAction(sa, null, Localizer.getInstance().getMessage("lblWouldYouLikeDestroy",(desc)), null)) {;
+        return;
```

Comment:
fix indentation

Reply:
should the valid be joined ? like `Lang.joinHomogenous` ?